src/util/luxon-utils.ts

```
@@ -51,3 +52,22 @@ export function assertValidDateTimeFormat(
+ * @param input - intervals to merge
+ * @returns an equivalent minimal set of Intervals without any intersections.
+ */
+export function mergeIntersecting(input: Interval<true>[]): Interval<true>[] {
+    return sortBy(input, ["start", "end"]).reduce<Interval<true>[]>((output, next) => {
+        const prevIndex = output.length - 1;
+        if (prevIndex >= 0 && output[prevIndex].intersection(next)) {
+            const union = output[prevIndex].union(next);
+            assertValidLuxonValue(union);
+            return output.toSpliced(prevIndex, 1, union);
+        } else {
+            return [...output, next];
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)